build-system/common/esbuild-babel.js

```diff
@@ -27,78 +86,129 @@ let transformCache;
  *   preSetup?: function():void,
  *   postLoad?: function():void,
  *   babelMaps?: Map<string, *>,
+ *   postCompileCaller?: string,
  * }} callbacks
  * @return {!Object}
  */
 function getEsbuildBabelPlugin(
   callerName,
   enableCache,
-  {preSetup = () => {}, postLoad = () => {}, babelMaps} = {}
+  {postCompileCaller, postLoad, preSetup} = {}
 ) {
-  /**
-   * @param {string} filename
-   * @param {string} contents
-   * @param {string} hash
-   * @param {Object} babelOptions
-   * @return {!Promise<!CacheMessageDef>}
-   */
-  async function transformContents(filename, contents, hash, babelOptions) {
-    if (enableCache) {
-      if (!transformCache) {
-        transformCache = new TransformCache('.babel-cache');
-      }
-      const cached = transformCache.get(hash);
-      if (cached) {
-        return cached;
-      }
-    }
-
-    debug('pre-babel', filename, contents);
-    const promise = babel
-      .transformAsync(contents, babelOptions)
-      .then((result) => {
-        const {code, map} = /** @type {!babel.BabelFileResult} */ (result);
-        debug('post-babel', filename, code, map);
-        return {filename, code: code || '', map};
-      });
-
-    if (enableCache) {
-      transformCache.set(hash, promise);
-    }
-
-    return promise.finally(postLoad);
-  }
-
+  const babelMaps = new Map();
   return {
     name: 'babel',
 
     async setup(build) {
-      preSetup();
+      preSetup?.();
+
+      const {initialOptions} = build;
+      const inlineSourcemap =
```

Comment:
why do we need this? what is setting it to `inline`? seems like we should just set it correctly from the getgo.